### ISSUE TITLE
Bundle android binary into nuget package

### ISSUE THREAD

**Issue by nshmyrev** [alphacep/vosk-api#654]
So it can run on Xamarin android.

**Comment by jnilsson89** [comment]
Do I use the language models the same way that I use them for standard C# applications? Or is it a different approach for Xamarin.Android? Thanks in advance!

**Comment by nshmyrev** [comment]
I believe in Android you need to unpack data to sdcard first like we are doing in plain Android. Otherwise it is the same.

**Comment by jnilsson89** [comment]
@nshmyrev Thank you for your answer! I've been trying to figure out how to use the model all morning but no luck. I put the model in the Download folder on my Android device and used the path to "Download/model/en-us" when specifying the  folder in new Model(path), but I get the exception "System.DllNotFoundException: 'libvosk assembly:<unknown assembly> type:<unknown type> member:(null)'". I've also tried with the .aar file for the English model with the same results.

**Comment by nshmyrev** [comment]
Hm, needs investigation. Did you unpack libvosk.so from aar? Does it find it at all? Can you access simple methods like SetLogLevel? Maybe some dll dependency is missing.

**Comment by jnilsson89** [comment]
I've got LibVosk among my dependencies and I can access the method you mentioned among others. Does the model need to be a .dll file or should it work to pass in the folder like in C#? Like this: Model model = new Vosk.Model("user/model/en-us"); but on the SD card instead. Thank you again for your response.

**Comment by philipag** [comment]
Just pinging if there is an ETA for this.  I tried compiling the android libvosk.so today but ran into quite a few problems until I gave up.  Seem that his is non-trivial and it would be really useful if the android libs were included in the Nuget.

**Comment by jnilsson89** [comment]
> Just pinging if there is an ETA for this. I tried compiling the android libvosk.so today but ran into quite a few problems until I gave up. Seem that his is non-trivial and it would be really useful if the android libs were included in the Nuget.

Do you get a problem loading a model like me?

**Comment by philipag** [comment]
> Do you get a problem loading a model like me?

No, everything works as expected.  I compiled libvosk.so version 3.,32 for Android using the instructions and manually added libvosk.so to the Xamarin Forms Android project which is using the Vosk nuget.

**Comment by philipag** [comment]
The instructions for installing and compiling are at [https://alphacephei.com/vosk/install](url)
They are a bit terse so you need to know what you're doing...

**Comment by jnilsson89** [comment]
> 

Thank you! Will try again.
